docs/external-ip-and-loadbalancer-ingress-in-shared-gateway-mode.md

```diff
@@ -0,0 +1,45 @@
+~~~
+External IPs
+If there are external IPs that route to one or more cluster nodes, Kubernetes Services can be exposed on those externalIPs. Traffic that ingresses into the cluster with the external IP (as destination IP), on the Service port, will be routed to one of the Service endpoints. externalIPs are not managed by Kubernetes and are the responsibility of the cluster administrator.
+~~~
+
+#### LoadBalancer Ingress (service.Status.LoadBalancer.Ingress)
+
+For a service's Status LoadBalancer Ingress field `service.Status.LoadBalancer.Ingress`, the aforementioned statement applies in exactly the same manner. Both External IP and `service.Status.LoadBalancer.Ingress` should behave the same from the network plugin's behavior, and it is the administrator's responsibility to get traffic for the VIPs into the cluster. 
+
+#### Implementation details
+
+OVN Kubernetes in Shared Gateway mode exposes External IPs and `service.Status.LoadBalancer.Ingress` VIPs as OVN load balancers on every node in the cluster. However, OVN Kubernetes will not answer to ARP requests to these VIP types, even if they reside on a node local subnet. This is because otherwise, every node in the cluster would answer with its own ARP reply to the same ARP request, leading to potential issues with stateful network flows that are tracked by conntrack. See the discussion in [https://github.com/ovn-org/ovn-kubernetes/issues/2407](https://github.com/ovn-org/ovn-kubernetes/issues/2407) for further details.
+
+In fact, OVN Kubernetes implements bypass rules for ARP requests to these VIP types on the external bridge (`br-ex` or `breth0` in most deployments). Any ARP request to such an IP that comes in from the physical port will bypass the OVN dataplane on purpose. 
+
+For implementation details, see: 
+* [https://github.com/ovn-org/ovn-kubernetes/blob/00925a6c64f57f03b2918eb48ff589c3417ddaa9/go-controller/pkg/node/gateway_shared_intf.go#L336](https://github.com/ovn-org/ovn-kubernetes/blob/00925a6c64f57f03b2918eb48ff589c3417ddaa9/go-controller/pkg/node/gateway_shared_intf.go#L336)
+* [https://github.com/ovn-org/ovn-kubernetes/blob/00925a6c64f57f03b2918eb48ff589c3417ddaa9/go-controller/pkg/node/gateway_shared_intf.go#L344](https://github.com/ovn-org/ovn-kubernetes/blob/00925a6c64f57f03b2918eb48ff589c3417ddaa9/go-controller/pkg/node/gateway_shared_intf.go#L344)
+
+For further discussion of the ARP bypass feature in the context of `service.Status.LoadBalancer.Ingress`, also see [https://github.com/ovn-org/ovn-kubernetes/pull/2394](https://github.com/ovn-org/ovn-kubernetes/pull/2394):
+~~~
+Currently address resolution requests (ARP/Neighbor solicitation) for a LoadBalancer IP
+are answered by all of the gateway routers in the cluster.
```

Comment:
```
are answered by all of the gateway routers in the cluster.
```

this is confusing, because on one side everywhere we are saying "we don't respond to arp requests for these VIPs". If this is for the L2 MetalLB mode like you have mentioned, maybe convert this para into

NOTE: Start with "In MetalLB L2 mode..."

Reply:
I'm going to drop the quote altogether, instead I just kept:
~~~
For further discussion of the ARP bypass feature in the context of `service.Status.LoadBalancer.Ingress`, also see [https://github.com/ovn-org/ovn-kubernetes/pull/2394](https://github.com/ovn-org/ovn-kubernetes/pull/2394).
~~~